src/app/Dashboard/LayoutTemplatePicker.tsx

```diff
@@ -0,0 +1,578 @@
+        <>
+          <StackItem>
+            <LayoutTemplateGroup
+              title={title}
+              templates={searchFilteredTemplates(sortedSearchFilteredTemplates)}
+              onTemplateSelect={onInnerTemplateSelect}
+              onTemplateDelete={onInnerTemplateDelete}
+            />
+          </StackItem>
+          <StackItem>
+            <Divider />
+          </StackItem>
+        </>
+      ) : (
+        <></>
+      );
+    },
+    [
+      searchFilteredTemplates,
+      selectedFilters,
+      selectedSort,
+      sortDirection,
+      onInnerTemplateSelect,
+      onInnerTemplateDelete,
+    ]
+  );
+
+  const RecentTemplates = React.useMemo(() => {
+    return recentTemplateRecords
+      .map((r) => {
+        const template = recordToLayoutTemplate(r, allTemplates);
+        if (template) {
+          return template;
+        } else {
+          console.error(`Could not find template for record ${r.name} with vendor ${r.vendor}`);
+          return undefined;
+        }
+      })
+      .filter((t) => t !== undefined) as LayoutTemplate[];
+  }, [recentTemplateRecords, allTemplates]);
+
+  React.useLayoutEffect(() => {
+    if (selectedTemplate) {
+      document
+        .querySelector('.layout-template-picker .catalog-tile-pf.featured')
+        ?.scrollIntoView({ behavior: 'smooth', block: 'center' });
+    }
+  }, [selectedTemplate]);
+
+  return (
+    <Drawer isExpanded={isDrawerExpanded} isInline>
+      <DrawerContent panelContent={panelContent}>
+        <DrawerContentBody>
+          <OuterScrollContainer>
+            <InnerScrollContainer>
+              <Toolbar isSticky clearAllFilters={onClearAllFilters}>
+                <ToolbarContent>
+                  <ToolbarToggleGroup toggleIcon={<FilterIcon />} breakpoint={'md'}>
+                    <ToolbarGroup variant="filter-group">
+                      <ToolbarItem variant="search-filter">
+                        <SearchAutocomplete values={allTemplates.map((t) => t.name)} onChange={onSearchChange} />
+                      </ToolbarItem>
+                      <ToolbarFilter chips={selectedFilters} deleteChip={onDeleteChip} categoryName="Category">
+                        <Select
+                          menuAppendTo={portalRoot}
+                          variant={SelectVariant.checkbox}
+                          aria-label="Select template category"
+                          onToggle={onFilterSelectToggle}
+                          onSelect={onFilterSelect}
+                          selections={selectedFilters}
+                          isOpen={isFilterSelectOpen}
+                          placeholderText="Template Type"
+                        >
+                          <SelectOption key="suggested" value={t('SUGGESTED', { ns: 'common' })} />
+                          <SelectOption key="cryostat" value="Cryostat" />
+                          <SelectOption key="user-submitted" value={t('USER_SUBMITTED', { ns: 'common' })} />
```

Comment:
The type should consistent?

![Screenshot from 2023-04-11 21-00-39](https://user-images.githubusercontent.com/68053619/231320367-0e0862c9-21d0-482a-8208-73852fc723c6.png)